src/Services/ExperimentationService.cs

```diff
@@ -35,4 +49,29 @@ public void AddExperimentalFeature(ExperimentalFeature experimentalFeature)
     {
         ExperimentalFeatures.Add(experimentalFeature);
     }
+
+    public bool IsExperimentationEnabled
+    {
+        get => _isExperimentationEnabled;
+
+        set
+        {
+            if (_isExperimentationEnabled != value)
+            {
+                _isExperimentationEnabled = value;
+
+                Task.Run(() =>

```

Comment:
Since this is synchronous and the task is not `await`ed, this will not prevent the UI from lagging, assuming that's the original concern. 

If the setter is a heavy operation, consider promoting it to a method.
  >if you are even considering providing an asynchronous version of an operation to avoid blocking the thread, it is very likely that the operation is too expensive to be a property -- [Source doc](https://learn.microsoft.com/en-us/previous-versions/dotnet/netframework-4.0/ms229054(v=vs.100)?redirectedfrom=MSDN)